p2p/nat/nat.go

```diff
@@ -85,6 +87,12 @@ func Parse(spec string) (Interface, error) {
 		return UPnP(), nil
 	case "pmp", "natpmp", "nat-pmp":
 		return PMP(ip), nil
+	case "stun":
+		var addr string
+		if len(after) > 1 {
+			addr = after
+		}
+		return NewSTUN(addr), nil
```

Comment:
`NewSTUN` should return an error -- the `addr` here is totally unverified, isn't it? `stun://x` would just silently return a stun address using `STUNDefaultServerAddr`

So this whole case could be 

```golang
case "stun":
    return NewSTUN(after)
```